server/monitor.go

```diff
@@ -2709,6 +2709,15 @@ func (s *Server) accountDetail(jsa *jsAccount, optStreams, optConsumers, optCfg
 	}
 	jsa.mu.RUnlock()
 
+	accStats := s.js.lookupAccount(acc)
```

Comment:
I am not familiar enough with "reserved" values to meaningfully comment, but we have a `jsa` object already and you are now looking up the `jsAccount` for the account of the `jsa` that you were given? They should be the same! (I know that there is a bug in the server that could lead to a race where a given registered account's jsAccount is not the one stored in `js.accounts[acc.Name]`, but not sure you were facing this issue).

So the limits that you check here should be done above for the given `jsa` and under the jsa.usageMu's lock.

Reply:
Thanks, will take a look now and add tests.

Reply:
Thanks, ok updated now.